mcdreforged/cli/cli_entry.py

```diff
@@ -157,3 +176,121 @@ def write(base_path: str, *, directory_only: bool):
 
 	writeln('Packed {} files/folders into "{}"'.format(file_counter, file_name))
 	writeln('Done')
+
+def init_plugin_workspace(path: str, pid: str, name: str, description: str, authors: str, link: str, resources: list[str], entrypoint: str, archive_name: str, *, quiet: bool = False):
+	writeln = print if not quiet else lambda *args, **kwargs: None
+	if quiet:
+		def ask(*args, default=None, **kwargs):
+			return default
+	else:
+		def ask(msg, *, default=None, skip=False):
+			r = input(('{} (enter to skip): ' if skip else '{}: ' if default is None else '{0} (default "{1}"): ').format(msg, default))
+			return default if len(r) == 0 else r
+
+	if path is None:
+		path = ask('Plugin work directory', default='.')
+
+	metafile = os.path.join(path, plugin_constant.PLUGIN_META_FILE)
+	if os.path.exists(metafile):
+		writeln('[ERROR] Meta file "{}" already exists'.format(metafile))
+		sys.exit(1)
+
+	if pid is None:
+		pid = ask('Id', default=os.path.basename(os.path.abspath(path))).strip()
+	if name is None:
+		name = ask('Name', default=pid).strip()
+	if description is None:
+		description = ask('Description', default='This is a plugin for MCDR').strip()
+	if authors is None:
+		authors = ask("Author(s), split with ':'")
+	if link is None:
+		link = ask('Main page link', skip=True)
+	if resources is None:
+		resources = ask("Resource(s), split with ':'", skip=True)
+	if entrypoint is None:
+		entrypoint = ask('Entry point', skip=True)
+	if archive_name is None:
+		archive_name = ask('Archive name', skip=True)
+
+	metadata = {
+		'id': pid,
+		'version': '1.0.0',
+		'name': name,
+		'description': description,
+		'dependencies': {
+			'mcdreforged': '>=2.0.0'
+		}
+	}
+	if entrypoint is not None:
+		metadata['entrypoint'] = entrypoint
+	if authors is not None:
+		metadata['author'] = list(set((a.strip() for a in authors.split(':'))))
+	if link is not None:
+		metadata['link'] = link
+	if resources is not None:
+		metadata['resources'] = list(set(resources.split(':')))
+	if archive_name is not None:
+		metadata['archive_name'] = archive_name
+
+	with open(metafile, 'w') as fd:
+		json.dump(metadata, fd, indent=4)
+		writeln('Created meta file "{}"'.format(metafile))
+
+	with open(os.path.join(path, 'requirements.txt'), 'w') as fd:
+		fd.write('# Add your python package requirements here, just like regular requirements.txt\n')
+
+	entry, point = (pid, '__init__.py') if entrypoint is None else entrypoint.split('.', 1)
+	if not os.path.exists(os.path.join(path, entry)):
+		os.mkdir(os.path.join(path, entry))
```

Comment:
suggest to use `os.makedirs`, in case of multiple nested directory

Reply:
I think it mustn't  
or can we use `entrypoint='dir1.dir2.dir3.file'`?

Reply:
It's possible to have the entrypoint to be like `plugin_id.dir1.dir2.dir3.entry` resulting in multiple nested directories are needed to be created. `os.mkdir` will fail in this situation